src/modules/workflowsettings/Elsa.WorkflowSettings/WorkflowSettingsOptionsBuilder.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using Elsa.Caching;
+using Elsa.WorkflowSettings.Abstractions.Persistence;
+using Elsa.WorkflowSettings.Options;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Elsa.WorkflowSettings
+{
+    public class WorkflowSettingsOptionsBuilder
+    {
+        public WorkflowSettingsOptionsBuilder(IServiceCollection services)
+        {
+            WorkflowSettingsOptions = new WorkflowSettingsOptions();
+            Services = services;
+            services.AddMemoryCache();
```

Comment:
It doesn't look like the Settings module uses caching itself. It's only evicting the workflow registry cache. If so, then you can remove this line.

Reply:
I guess we can remove the next line either?
services.AddSingleton<ICacheSignal, CacheSignal>();

Reply:
Strictly speaking yes, since this service is registered by Elsa Core. However, since this module relies on this service too, it makes sense to me to keep it. But instead of adding another registration, maybe better to do TryAddSingleton. This way, it only creates a new entry if one doesn't already exist.